.github/workflows/main.yml

```diff
@@ -0,0 +1,43 @@
+# Adapted from https://github.com/ocaml/setup-ocaml
+#
+name: ci
+
+# The lints should be only run when packages are added
+on:
+  pull_request:
+    paths:
+      - packages/
+  push:
+    paths:
+      - packages/
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout tree
+        uses: actions/checkout@v4
+        with:
+          # We need to fetch the history so we can compare against master
+          fetch-depth: 0
+
+      - name: Install GNU prallel
+        run: sudo apt-get install -y parallel
+
+      - name: Set-up OCaml
+        uses: ocaml/setup-ocaml@v3
+        with:
+          ocaml-compiler: 5
+
+      - name: Install opam-ci-check
+        run: opam pin opam-ci-check https://github.com/shonfeder/opam-repo-ci.git#411/archive-lint
+
+      # Get the names of the packages being added to the archive, and feed these
+      # to opam-ci-check
+      - name: Lint newly archived packages
+        run: |
+          git diff --name-only origin/main packages \
+          | sed 's:.*/\(.*\)/opam:\1:' \
+          | xargs --max-lines=1 --max-procs=4 opam exec -- opam-ci-check lint -r . --checks=archive-repo \
+          | grep "Error in"
```

Comment:
I'm a bit concerned about the exit code: `grep` returns 0 if it matched something ; and non-zero if there was no match.

In our CI, we would like to return 0 on success?

Reply:
we can look into this later. I'll go ahead and merge this PR.

Reply:
Oh you are definitely right about that. This will mask the exit codes.

I'll fix the linter so it has a quiet mode so we don't need this filter, and update here in a bit. Thanks for the review!

Reply:
Ah I forgot that github actions runs this stuff with `pipefail` set, so we are actually fine here w/r/t to exit codes. Having cleaner output from to tool is desirable per se, but I think the current behavior should be correct w/r/t exits.